src/pages/post/Post.tsx

```diff
@@ -0,0 +1,222 @@
+          <NavigationMenuList className="flex flex-col flex-wrap md:flex-row">
+            <NavigationMenuItem className="mx-4 w-40">
+              <NavigationMenuLink className={navigationMenuTriggerStyle()}>
+                Home
+              </NavigationMenuLink>
+            </NavigationMenuItem>
+            <NavigationMenuItem className="mx-4 w-40">
+              <NavigationMenuLink className={navigationMenuTriggerStyle()}>
+                About
+              </NavigationMenuLink>
+            </NavigationMenuItem>
+            <NavigationMenuItem className="mx-4 w-40">
+              <NavigationMenuLink className={navigationMenuTriggerStyle()}>
+                Profile
+              </NavigationMenuLink>
+            </NavigationMenuItem>
+          </NavigationMenuList>
+        </NavigationMenu>
+      </div>
+      <div className="col-span-1 flex justify-evenly align-middle">
+        <Link to="">
+          <Avatar>
+            <AvatarImage src="random_link" />
+            <AvatarFallback>
+              <Icon icon="material-symbols-light:face-6" className="text-2xl" />
+            </AvatarFallback>
+          </Avatar>
+        </Link>
+        <Button className="flex justify-evenly align-middle">
+          <Icon icon="material-symbols-light:login" className="text-2xl" />
+        </Button>
+      </div>
+    </div>
+  );
+};
+
+export const Post: React.FC = () => {
+  // TODO: implement providers
+  // TODO: implement error page if user or post is not found
+  const currentPost = posts.filter((post) => post.id === "post_1")[0];
+
+  const postAuthor = users.filter(
+    (user) => user.id === currentPost?.user_id
+  )[0];
+
+  return (
+    <div className=" overflow-hidden bg-white">
+      <NavBar />
+      <div
+        id="post-title"
+        className="mb-4 mt-40 flex w-screen justify-center p-11 md:mt-20"
+      >
+        <p
+          className="text-6xl"
+          style={{
+            fontFamily: "Oswald, sans-serif",
+            fontOpticalSizing: "auto",
+            fontWeight: 618,
+            fontStyle: "normal",
+          }}
+        >
+          {currentPost?.title}
+        </p>
+      </div>
+      <div className="flex w-screen justify-center">
+        <div id="blog-info" className="grid w-[500px] grid-cols-3">
+          <div id="user-info" className="flex items-center justify-center">
+            <Icon icon="material-symbols-light:face-6" className="text-2xl" />
+            <span className="mx-1">
+              by <strong>{postAuthor.first_name}</strong>
+            </span>
+          </div>
+          <div
+            id="read-duration-info"
+            className="flex items-center justify-center"
+          >
+            <Icon
+              icon="material-symbols-light:nest-clock-farsight-analog-outline"
+              className="text-2xl"
+            />
+            <span className="mx-1">{`${calculateReadTime(currentPost?.content || "").minutes} min read`}</span>
```

Comment:
add default value on text param of calc... fn

Reply:
already calculated by the calculateReadTime function

Reply:
function calculateReadTime(text = "") 
to always handle nullish input to calculateReadTime by default

Reply:
calculateReadTime -> calculateReadDuration